### ISSUE TITLE
Directional light doesn't illuminate entire scene

### ISSUE THREAD

**Issue by Brfy** [Bigfoot71/r3d-legacy#1]
![image](https://github.com/user-attachments/assets/e6ea6507-80f7-41d5-9695-5ef51074b765)
Here I have attached an image of a very simple ball rendering test, and as you can see, only a fraction of the balls are lit properly. Only the balls in a small section get shadows.

Here is the code used for the directional light:
```
R3D_Light sun = R3D_CreateLight(R3D_DIRLIGHT, 1024);
R3D_SetLightDirection(sun, Vector3Normalize((Vector3) { -1, -1, -1 }));
R3D_SetLightActive(sun, true);
```

**Comment by Bigfoot71** [comment]
Yes, I need to implement a CSM or another technique for directional lights, it's written in the `TODO` file. This will be the next thing I add. I should be able to do it in a few days.

By the way, this issue should be adjustable by specifying a position for the directional lights. It's a bit counterintuitive, but the position is used as the viewpoint for shadow mapping.

Try something like:
```c
R3D_Light sun = R3D_CreateLight(R3D_DIRLIGHT, 1024);
R3D_SetLightPositionTarget(sun, (Vector3) { 10, 10, 10 }, (Vector3) { 0 });
R3D_SetLightActive(sun, true);
```
